src/makeInfinityAdder.js

```diff
@@ -4,7 +4,27 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let currentSum = 0;
+
+  const ADDER = (num) => {
+    if (num !== undefined) {
+      currentSum += num;
+
+      return ADDER;
+    } else {
+      sum += currentSum;
```

Comment:
```suggestion
    }
      sum += currentSum;
```
else is redundant here